lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/driver/AzureBatchEvaluatorShimConfigurationProvider.java

```diff
@@ -32,10 +38,23 @@
 public class AzureBatchEvaluatorShimConfigurationProvider {
 
   private final RemoteManager remoteManager;
+  private final TcpPortProvider portProvider;
+  private final LocalAddressProvider localAddressProvider;
+  private final Set<String> tcpPortList;
 
   @Inject
-  AzureBatchEvaluatorShimConfigurationProvider(final RemoteManager remoteManager) {
+  AzureBatchEvaluatorShimConfigurationProvider(
+      @Parameter(TcpPortList.class) final Set<Integer> tcpPortList,
+      final RemoteManager remoteManager,
+      final LocalAddressProvider localAddressProvider,
+      final TcpPortProvider portProvider) {
     this.remoteManager = remoteManager;
+    this.portProvider = portProvider;
+    this.localAddressProvider = localAddressProvider;
+    this.tcpPortList = new HashSet<String>(tcpPortList.size());
+    for (int port: tcpPortList) {
+      this.tcpPortList.add(Integer.toString(port));
+    }
```

Comment:
Another conversion from Integer to String.

Reply:
setMultiple takes a set of strings. That is why.

---
In reply to: [199969006](https://github.com/sharathmalladi/reef/pull/6#discussion_r199969006) [](ancestors = 199969006)